### PR TITLE
fix support blob compile error: (eg: query<tuple<A,B>>())

### PR DESCRIPTION
**Pull request by lemon19900815**

### CODE REVIEW

**Comment by qicosmos** [comment]
could you also add test case?

**Comment by lemon19900815** [comment]
now i added test code.

File: include/mysql.hpp [hunk 1]
```diff
@@ -292,7 +295,7 @@ class mysql {
                 using V =
                     std::remove_reference_t<decltype(std::declval<W>().*ele)>;
                 if constexpr (std::is_arithmetic_v<V>) {
-                  item.*ele = *(V *)(&(*it)[0]);
+                  // item.*ele = *(V *)(&(*it)[0]);
```
Comment (qicosmos) [inline include/mysql.hpp:298]: why delete this code?

File: include/mysql.hpp [hunk 2]
```diff
@@ -224,13 +224,16 @@ class mysql {
...
-              else if constexpr (std::is_same_v<blob, U>) {
+              else if constexpr (std::is_same_v<blob, V>) {
                 std::vector<char> tmp(65536, 0);
                 mp.emplace_back(std::move(tmp));
                 param_binds[index].buffer_type = MYSQL_TYPE_BLOB;
                 param_binds[index].buffer = &(mp.back()[0]);
                 param_binds[index].buffer_length = 65536;
               }
+              else {
+                std::cout << typeid(V).name() << std::endl;
```
Comment (qicosmos) [inline include/mysql.hpp]: use static_assert,
change to ``` static_assert(!sizeof(V), "not support this type"); ```

File: include/mysql.hpp [hunk 3]
```diff
@@ -276,7 +279,7 @@ class mysql {
           [&mp, &it, &column, this](auto &item, auto /*i*/) {
             using W = std::remove_reference_t<decltype(item)>;
             if constexpr (std::is_arithmetic_v<W>) {
-              return;
+              // return; // don't return, need ++column at end.
```
Comment (qicosmos) [inline include/mysql.hpp:282]: we don't need to deal with arithmetic type?

Comment (lemon19900815) [reply include/mysql.hpp:282]: the column is blob location, if there is an or more arithmetic field before blob, the location is not correct.

File: include/mysql.hpp (hunk 2 above)
Comment (lemon19900815) [reply include/mysql.hpp]: ok

File: include/mysql.hpp (hunk 1 above)
Comment (lemon19900815) [reply include/mysql.hpp:298]: i'm not sure for this. but i can't get correct result with this code. i think bind_param for the arithmetic field address, there don't need it, so i delete it.

File: include/mysql.hpp (hunk 3 above)
Comment (qicosmos) [reply include/mysql.hpp:282]: So there is no arithmetic type here? we can add a static_assert to make sure no arithmetic here.

File: include/mysql.hpp (hunk 1 above)
Comment (qicosmos) [reply include/mysql.hpp:298]: If we don't need deal with arithmetic type, we can add a static_assert to make sure no arithmetic here?

Comment (qicosmos) [reply include/mysql.hpp:298]: could you check if there is a bug here

Comment (lemon19900815) [reply include/mysql.hpp:298]: 图片
mysql query result store as arithmetic type use the item address it self(place 1), not use the `mp`(std::list<std::vector<char>>, place 2), but on the result fetch, use `it`(the mp iterator) to set the arithmetic type data, i think it's a bug.

**Pull request merged**